### ISSUE TITLE
Fix walk-forward data insufficiency errors and make train_window_days configurable

### ISSUE THREAD

**Issue by alexoms** [OptimalMatch/temporal-trading-agents#109]
## Problem Summary

During auto-optimization with 28 parallel workers, we encountered two types of errors:

### Error 1: File Corruption ✅ FIXED
```
PytorchStreamReader failed reading file data/17: file read failed
```
**Root Cause**: Multiple workers fine-tuning the same model and saving to the same cache file simultaneously
**Solution**: Only save fresh models to cache, keep fine-tuned models in memory only

### Error 2: Data Preparation Failure ⚠️ IN PROGRESS
```
ERROR: Model training failed: __len__() should return >= 0
```
**Root Cause**: Walk-forward windows too small for model requirements
- Walk-forward uses 365-day training window (hardcoded)
- After 70/15/15 split: ~255 train, ~55 val, ~55 test samples  
- Models with lookback=60 + forecast=14 need 74 samples
- Validation set has only 55 samples < 74 needed ❌

### 1. Made `train_window_days` Configurable

**Backend (`backend/models.py`):**
```python
class AutoOptimizeConfig(BaseModel):
    ...
    train_window_days: int = 547  # New field with increased default
    # 547 days → ~383 train, ~82 val, ~82 test samples
    # Supports lookback up to 60 + forecast 14 = 74 samples
```

**Backend (`backend/auto_optimize_manager.py`):**
- Updated all 4 walk-forward config locations to use `run.config.train_window_days`
- Removed hardcoded 365-day value

**Frontend (`frontend/src/pages/AutoOptimizePage.jsx`):**
- Added `train_window_days: 547` to config state
- Added UI input field with label and help text
- Users can now customize window size (200-1000 days)

### 2. Graceful Fallback for Insufficient Data

**Training Code (`examples/crypto_ensemble_forecast.py`):**
```python
# Validate data before creating datasets
if len(train_data) < min_samples_needed or len(val_data) < min_samples_needed:
    if use_cached:
        print("⚠️  Insufficient data for fine-tuning:")
        print(f"   Training: {len(train_data)} samples (need {min_samples_needed})")
        print(f"   Validation: {len(val_data)} samples (need {min_samples_needed})")
        print("💡 Skipping fine-tuning - using cached model as-is")
        print(f"   Tip: Increase train_window_days or reduce lookback")
        
        # Return cached model without fine-tuning
        return {..., 'training_type': 'cached-only'}
    else:
        # Clear error message with actionable advice
        raise ValueError(...)
```

### 3. File Corruption Fix (Already Deployed)

**Model Cache (`examples/crypto_ensemble_forecast.py`):**
```python
# Save to cache only for fresh models, not fine-tuned ones
if use_cache and training_type == "fresh":
    model_cache.save(...)  # Save base models
elif use_cache and training_type == "fine-tuned":
    print("💡 Fine-tuned model kept in memory only")  # No save
```

### UI Date Range vs Train Window

**Important Clarification:**
- **UI Date Range** (start_date → end_date): Controls total backtest period (e.g., 2 years)
- **Train Window Days**: Controls sliding window size for walk-forward (default: 547 days)

These are **separate** parameters!

**Example:**
- UI: Jan 2022 → Jan 2024 (2 years total backtest)
- Train Window: 547 days (each sliding window)
- Walk-forward: Slides 547-day window through the 2-year period
- Tests on 63-day periods, retrains every 21 days

### New UI Field

Screenshot of new field in Auto-Optimize page:
```
Walk-Forward Training Window (days)  [Default: 547 (~1.5 years)]
[ 547 ]  ← User can adjust 200-1000

Sliding window size for walk-forward backtesting. 
Increase if you see "insufficient data" warnings.
```

## Configuration Recommendations

| Scenario | train_window_days | Notes |
|----------|-------------------|-------|
| Default (safe) | 547 | Supports lookback 60 + forecast 14 |
| Long backtests | 730+ | More historical context |
| Short models | 365 | If using lookback < 45 |
| Very long models | 800-1000 | If using lookback > 60 |

### Before (Cryptic)
```
ERROR: Model training failed: __len__() should return >= 0
```

### After (Actionable)
```
⚠️  Insufficient data for fine-tuning:
   Training: 255 samples (need 74)
   Validation: 55 samples (need 74)
💡 Skipping fine-tuning - using cached model as-is
   Tip: Increase train_window_days (currently 365 total samples) or reduce lookback
```

## Testing Performed

- ✅ File corruption errors eliminated with 28 parallel workers
- ✅ Fine-tuned models no longer saved (avoid race conditions)
- ✅ Graceful fallback when data insufficient
- ✅ Clear user messaging when fine-tuning skipped
- ✅ UI exposes train_window_days parameter
- ✅ Default 547 days works with standard models (lookback 60, forecast 14)

## Files Modified

**Backend:**
- `backend/models.py` - Added `train_window_days` to AutoOptimizeConfig
- `backend/auto_optimize_manager.py` - Use configurable value (4 locations)
- `examples/crypto_ensemble_forecast.py` - Graceful fallback + better error messages

**Frontend:**
- `frontend/src/pages/AutoOptimizePage.jsx` - Added UI field for train_window_days

## Related Issues

- Resolves walk-forward data insufficiency errors with high parallelization (28 workers)
- Resolves file corruption when multiple workers fine-tune same model
- Improves user visibility into walk-forward configuration

## Migration Notes

**Existing auto-optimize configurations:**
- Will automatically use new default: 547 days (safe, prevents errors)
- Users can reduce to 365 if desired (old default)
- No breaking changes

**Recommended Actions:**
1. Review existing auto-optimize runs - if seeing errors, increase train_window_days
2. For new backtests with long lookbacks (>60), use 730+ days
3. Monitor logs for "💡 Fine-tuned model kept in memory" messages (confirms fix working)
